src/include/units/bits/external/type_traits.h

```diff
@@ -58,32 +59,21 @@ template<class T, class U>
 using is_same = std::bool_constant<is_same_v<T, U>>;
 
 // is_specialization_of
-namespace detail {
-
 template<typename T, template<typename...> typename Type>
-inline constexpr bool is_specialization_of_impl = false;
+inline constexpr bool is_specialization_of = false;
 
 template<typename... Params, template<typename...> typename Type>
-inline constexpr bool is_specialization_of_impl<Type<Params...>, Type> = true;
-
-}  // namespace detail
-
-template<typename T, template<typename...> typename Type>
-inline constexpr bool is_specialization_of = detail::is_specialization_of_impl<T, Type>;
+inline constexpr bool is_specialization_of<Type<Params...>, Type> = true;
 
 // is_derived_from_specialization_of
 namespace detail {
 
-template<template<typename...> typename Type>
-struct is_derived_from_specialization_of_impl {
-  template<typename... Params>
-  static constexpr std::true_type check_base(const Type<Params...>&);
-  static constexpr std::false_type check_base(...);
-};
+template<template<typename...> typename Type, typename... Params>
+void to_base_specialization_of(const volatile Type<Params...>*);
 
 }  // namespace detail
 
 template<typename T, template<typename...> typename Type>
-inline constexpr bool is_derived_from_specialization_of = decltype(detail::is_derived_from_specialization_of_impl<Type>::check_base(std::declval<T>()))::value;
+concept is_derived_from_specialization_of = requires { detail::to_base_specialization_of<Type>(std::declval<const volatile T*>()); };
```

Comment:
I think we should be consistent here. Either both of them are traits or both of them are concepts.

Reply:
Right. The other two should be concepts, but that's only to make it work on MSVC. Otherwise they'd be variables.

Reply:
In such a case maybe let's #ifdef them to be concepts on MSVC and leave variables for GCC or add TODO comment to fix it when MSVC will start to support this. Otherwise, we will forget to clean this up.